### ISSUE TITLE
Bug with paste-retaining-clones (and it's undo/redo as side-effect?)

### ISSUE THREAD

**Issue by boltex** [leo-editor/leo-editor#3446]
Closed via PR #3468.

See PR #3449. The failing unit test shows that gnxs are correct, but `v.parents` is bizarre.

When cut/copy and then paste as **single node** as clone no bugs occur. 

**But**, when copying a suboutline that contains a clone of a node outside of itself, and pasting it somewhere else, both those nodes that should be linked as clone again after that paste-as-clone command **are not clones of themselves anymore**. 

_all of those operation, clone, move the clone outside the subtree and cut, do work fine and can be undone at any point no problem, but not that last one: **paste-as-clone**_

![image](https://github.com/leo-editor/leo-editor/assets/847716/ec1d6437-29a3-4ed5-997f-d52339301331)
Here I cut **cc** and paste-as-clone again. **cc**'s **child1** is surprisingly not a clone!
![image](https://github.com/leo-editor/leo-editor/assets/847716/7471636c-4c5b-419f-8748-dc3cdd62f7a5)

And then trying to **undo that operation gives a bunch of red text** in the log pane.

( **On the other hand, 'paste-as-template'  does work! and undo/redoes properly to!** - IF I understand correclty: this command just honors clones contained in it's subtree being pasted, but does not clone the top node being pasted, where paste-as-clone also makes a clone of the top node being pasted) 

I'll make more tests with paste-as-template to see if it honors body-content and ua changes of what is being pasted and report in the comments below this issue... 

```
Traceback (most recent call last):
  File "C:\prog\leo-editor\leo\core\leoGlobals.py", line 304, in new_cmd_wrapper
    func(self, event=event)
  File "C:\prog\leo-editor\leo\core\leoUndo.py", line 1611, in undo
    u.undoHelper()
  File "C:\prog\leo-editor\leo\core\leoUndo.py", line 1819, in undoInsertNode
    c.deleteOutline()
  File "C:\prog\leo-editor\leo\commands\commanderOutlineCommands.py", line 1105, in deleteOutline
    p.doDelete(newNode)
  File "C:\prog\leo-editor\leo\core\leoNodes.py", line 1556, in doDelete
    p._unlink()
  File "C:\prog\leo-editor\leo\core\leoNodes.py", line 1196, in _unlink
    child._cutLink(n, parent_v)
  File "C:\prog\leo-editor\leo\core\leoNodes.py", line 2635, in _cutLink
    child._cutParentLinks(parent=v)
  File "C:\prog\leo-editor\leo\core\leoNodes.py", line 2640, in _cutParentLinks
    v.parents.remove(parent)
ValueError: list.remove(x): x not in list
```
**Addendum**
Logic of paste-as-clone that I suggest should apply: 

The loop that looks at the pasted content going from node to node in the natural tree order (to create the new outline being pasted) should check the current commander's gnx's of nodes it's trying to instantiate, and use existing VNodes when they already exist in the commander. (thus creating clones) Also, the pasted nodes body content - AND - ua's should replace existing content of those VNodes... 
(AND their children's array of vnodes.... obviously, hehe! so that the new tree structure is created as a pasted outline. This last part was probably already the case. Just noting this here to make sure.) 

Thanks!

**Comment by edreamleo** [comment]
Thanks for this report. I'll take a look at the logic immediately.

**Comment by edreamleo** [comment]
Closed via PR #3468.
